icu4c/source/i18n/unitconverter.cpp

```diff
@@ -5,14 +5,73 @@
 
 #if !UCONFIG_NO_FORMATTING
 
+#include "charstr.h"
+#include "measunit_impl.h"
+#include "unicode/errorcode.h"
+#include "unicode/measunit.h"
+#include "unicode/stringpiece.h"
+#include "unitconverter.h"
+
 U_NAMESPACE_BEGIN
 
+namespace {
+/**
+ * Extracts the compound base unit of a compound unit (`source`). For example, if the source unit is
+ * `square-mile-per-hour`, the compound base unit will be `square-meter-per-second`
+ */
+MeasureUnit extractCompoundBaseUnit(const MeasureUnit &source, const ConversionRates &conversionRates,
+                                    UErrorCode &status) {
+    MeasureUnit result;
+    int32_t count;
+    const auto singleUnits = source.splitToSingleUnits(count, status);
+    if (U_FAILURE(status)) return result;
```

Comment:
Stylistic point: in my code review, we've settled on adding braces around these return statements:
    if (...) { return result; }

This based on the coding guidelines under heading "if() {...} and Loop Bodies" : 
http://userguide.icu-project.org/dev/codingguidelines#TOC-C-and-C-Type-and-Format-Convention-Guidelines
I settled on usually keeping it on one line if it's just a "return" statement, whereas giving the body its own line if it's anything more complicated. (Pity I couldn't figure out how to get clang-format to be more aligned with this style compromise.)

Reply:
Searcthing for ") return" finds examples like these. (Including an old line of mine where I didn't follow this style yet.)

Reply:
I'm dropping this for now: we can do more stylistic fixes later, and "no braces" is seen in existing code too, anyway.